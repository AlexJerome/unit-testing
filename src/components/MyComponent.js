import React from "react";

export default class MyComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = { count: 0 };
  }
  render() {
    return (
      <div>
        <h1>Compteur</h1>
        <h2>Valueur du compteur : {this.state.count}</h2>
        <button
          id="increment"
          onClick={() => this.setState({ count: this.state.count + 1 })}
        >
          + 1
        </button>
        <button
          id="decrement"
          onClick={() => this.setState({ count: this.state.count - 1 })}
        >
          -1
        </button>
      </div>
    );
  }
}
