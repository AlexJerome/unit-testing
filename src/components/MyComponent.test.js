import React from "react";
import { shallow, mount } from "enzyme";
import MyComponent from "./MyComponent";

describe("Test MyComponent", () => {
  it("should render correctly", () => {
    const component = shallow(<MyComponent />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("should increment when button +1 pressed", () => {
    const component = mount(<MyComponent />);
    const prevValueOfCount = component.state("count");
    component.find("button#increment").simulate("click");
    expect(component.state("count")).toEqual(prevValueOfCount + 1);
    component.unmount();
  });

  it("should decrement when button -1 pressed", () => {
    const component = mount(<MyComponent />);
    const prevValueOfCount = component.state("count");
    component.find("button#decrement").simulate("click");
    expect(component.state("count")).toEqual(prevValueOfCount - 1);
    component.unmount();
  });
});
