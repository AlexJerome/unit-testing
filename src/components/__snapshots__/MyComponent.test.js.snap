// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyComponent should render correctly 1`] = `
<div>
  <h1>
    Compteur
  </h1>
  <h2>
    Valueur du compteur : 
    0
  </h2>
  <button
    id="increment"
    onClick={[Function]}
  >
    + 1
  </button>
  <button
    id="decrement"
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
